File: guests/java/dev/kaya/milestone2kt/Split.java
package dev.kaya.milestone2kt;

import dev.kaya.KayaApp;

/**
 * The split conformance scene from the JVM — adaptive list-detail via
 * the chain spelling: listDetail rides the window chain, pushEntry
 * chains the entry's props, mountIn presents its root, and onPopped
 * hears the user's native pop.
 *
 * <p>The guest asks for the presentation ONCE and then does nothing
 * adaptive ever again. Everything after that is the platform
 * re-deciding as the size class changes: an app does not write two
 * layouts and pick one, and there is no prop for WHICH way it
 * presents. Nothing here is split-specific except that one prop.
 *
 * <p>TWO scripts drive this ONE app: split resizes and names the
 * presentation on each side, listdetail asserts the bare invariant at
 * whatever width its host gives. See guests/rust/split.rs,
 * tools/scenes/split.steps and tools/scenes/listdetail.steps.
 */
final class Split {
    private static final long DETAIL = 7;

    static void app() {
        KayaApp app = new KayaApp();

        KayaApp.Signal<String> status = app.build(tx -> {
            // The one adaptive declaration in the whole guest.
            tx.window(0).title("split").listDetail(true);
            KayaApp.Signal<String> s = tx.signal("list pane");
            tx.mount(tx.column(() -> {
                // Authored ids so the REAL-TREE read can address
                // these: an index read passes whether or not anything
                // reached the screen, which is the gap that let a
                // non-rendering split arm look green.
                tx.label(s).a11yId("list"); // label#0
                tx.button("open detail", inner -> { // button#0
                    // The popped handler rides the push, per-entry —
                    // the onResult precedent, unchanged by the split.
                    long entry = inner.pushEntry(DETAIL)
                            .title("detail")
                            // Retention: the base root took this write
                            // while the detail was up, on a regular
                            // window where it was VISIBLE the whole
                            // time.
                            .onPopped(tx2 -> tx2.write(s, "popped detail"))
                            .id();
                    KayaApp.Widget pane = inner.column(() -> {
                        KayaApp.Signal<String> caption = inner.signal("detail pane");
                        inner.label(caption).a11yId("detail");
                    });
                    inner.mountIn(entry, pane);
                });
            }));
            return s;
        });

        // status is captured by the handlers above; keep the local
        // alive for symmetry with the other scenes.
        if (status == null) throw new IllegalStateException();

        app.dispatchLoop();
    }

    private Split() {}
}

File: guests/go/split/split.go
// The split scene, driven by TWO scripts (tools/scenes/split.steps and
// listdetail.steps): the presentation is asked for ONCE and never again.
package split

import (
	kaya "dev.kaya/bindings/go"
)

const detail = 7

func App() *kaya.App {
	app := kaya.NewApp()

	var status kaya.Signal[string]
	app.Build(func(tx *kaya.Tx) {
		tx.Window(0).Title("split").Panes(2)
		status = tx.Signal("list pane")

		tx.Mount(tx.Column(func() {
			// Authored ids: an index read passes for an empty arm.
			tx.Label(status).A11yID("list") // label#0
			tx.Button("open detail", func(tx *kaya.Tx) { // button#0
				entry := tx.PushEntry(detail).
					Title("detail").
					OnPopped(func(tx *kaya.Tx) {
						tx.Write(status, "popped detail")
					}).
					Id()
				pane := tx.Column(func() {
					caption := tx.Signal("detail pane")
					tx.Label(caption).A11yID("detail")
				})
				tx.MountIn(entry, pane)
			})
		}))
	})

	return app
}
